lib/logproto/logproto-server.c

```diff
@@ -122,12 +122,20 @@ log_proto_server_init(LogProtoServer *self, LogTransport *transport, const LogPr
   self->transport = transport;
 }
 
-void
+gboolean
 log_proto_server_options_set_encoding(LogProtoServerOptions *self, const gchar *encoding)
 {
-  if (self->encoding)
-    g_free(self->encoding);
+  GIConv convert;
+
+  g_free(self->encoding);
   self->encoding = g_strdup(encoding);
+
+  /* validate encoding */
+  convert = g_iconv_open("utf8", encoding);
```

Comment:
I think the problem is that we want to open an utf8 to utf8 encoding and
that fails.
On May 10, 2016 8:02 AM, "Budai Laszlo" <notifications@github.com> wrote:

> In lib/logproto/logproto-server.c
> <https://github.com/balabit/syslog-ng/pull/797#discussion_r62616419>:
>
> >    self->encoding = g_strdup(encoding);
> > +
> > +  /* validate encoding */
> > +  convert = g_iconv_open("utf8", encoding);
>
> @bazsi <https://github.com/bazsi>:
>
> Here is the 'internal' 'bug':
>
> seems utf-8 encoding does not work.
> [2016-05-03T11:43:11.043665] Module loaded and initialized successfully; module='afprog'
> Error parsing afprog, unknown encoding UTF-8 in /opt/syslog-ng/etc/syslog-ng.conf at line 6, column 88:
> program("/var/testdb_working_dir/90ee8394-0d84-402d-933b-c7997a6dc3e7.sh" encoding(UTF-8));
> config:
> @version: 3.8
> options
> { threaded(yes); }
> ;
> source s_program_37b57ebaaa6a460085d2d35d6f7d56ec
> { program("/var/testdb_working_dir/644aefb0-504c-429f-8022-ce3150b60b75.sh" encoding(UTF-8)); }
> ;
> destination d_file_a3219214c5e541d5a18b0e9ca3eedee2
> { file("/var/testdb_working_dir/81525975-13e5-4a41-8e9e-1bbd7cdc3d25.txt"); }
> ;
> log
> { source(s_program_37b57ebaaa6a460085d2d35d6f7d56ec); destination(d_file_a3219214c5e541d5a18b0e9ca3eedee2); flags(flow-control); }
> ;
>
> @bkil-syslogng <https://github.com/bkil-syslogng>:
> To be honest, the way you are handle this issue is not the best...
> We found something during internal tests and *before* the validation and
> tests you are
> blaming an already-merged PR (without sharing all the information you
> have).
>
> Even if the bug is valid and your fix resolves the issue, we have
> a well defined process that you should follow. When we found 'something'
> and we validated that this is a bug then we open a corresponding github
> issue (or, it is also fine if you open the github issue before the
> validation), the point is that we share *everything* we know about the
> issue and when everyone has the chance at being fully informed then we can
> start the discussion.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/balabit/syslog-ng/pull/797/files/35c68349ce999d0318e59abcb3cf4cccef630c81..3fa60288548b54bcfad7b42deb145ef2eafa029c#r62616419>
>